### DISCUSSION TITLE
How to get sleep working like it was before installing ghelper?

### DISCUSSION THREAD

**Discussion by daigle-grid-engineering** [seerge/g-helper#674] [Q&A]
I have a 2022 G14 and I've noticed that my laptop now has different behavior during sleep.

The power LED used to have a heartbeat while sleeping but that seems to not be the case anymore. Additionally I can hear the fans kick on. Has anyone else experienced this?

**Comment by seerge** [comment]
@3aughtmaxon your sleep is controlled by windows :) and it's power plan. Try use default (balanced) power plan and and reset it to defaults

**Comment by daigle-grid-engineering** [comment]
That may be true, but the installation of ghelper and uninstallation of Armoury Crate is what is triggering the issue. Will investigate and report back. I've done this process (uninstall/reinstall AC and switch to ghelper) twice.

**Reply by seerge** [reply]
G-helper does absolutely nothing that could affect windows sleep anyhow :) however installing/uninstalling armoury does (it adds/removes windows power plans related to AC)

**Comment by steverand** [comment]
I found this while checking if anyone else has had this issue. I'll start another thread with the below:
When Ghelper checkbox "auto adjust windows power modes" is on, my laptop does the same thing. It freezes (according to the logs) when it tries to go into Windows Modern Standby after being in sleep for some time. It took me weeks of Googling and experimenting with registry entries but as soon as I unchecked that and deleted  a reg entry (aoac override or similar) it worked perfectly again. Powercfg output says my laptop does not support Windows Modern Standby.
Now with the latest update (0.110) it happens again. I had to roll back to 0.100. I think the latest update took the option away, so i'm assuming it's doing it automatically?

Anyway, no AC, only "asus optimization" service enabled, running 0.100 and no auto adjust power mode works fine for my A17FA706IC.

**Reply by seerge** [reply]
@steverand hello, power modes are just a setting in windows (check screenshot from release page). Does changing it manually inside windows also cause your system to freeze? I would check whole system for corruption in this case.

https://support.microsoft.com/en-au/topic/use-the-system-file-checker-tool-to-repair-missing-or-corrupted-system-files-79aa86cb-ca52-166a-92a3-966e85d4094e

Anyway, the only difference in 0.110 is that now you can actually select (via fans+power) what power mode to set with each preset. If you don't want it to change - just set **same** power mode for all 3 presets :) (the one you used before)

![Screenshot 2023-07-14 123413](https://github.com/seerge/g-helper/assets/5920850/dfec3b29-0ecf-46d8-95e6-22cc42a862a8)

**Reply by steverand** [reply]
Setting modes in Windows is no issue. I can have different plans for battery/AC with no issue. I only use "Balanced" and never change it.
It only tries to enter Modern Standby when the GHelper option to adjust Windows power modes is checked. I wasn't 100% sure this was the whole problem but then as soon as I installed the last update, it did it again, with nothing else changed on my laptop. Since the option wasn't there anymore, I rolled back to 0.100 and immediately it sleeps without issue.

**Reply by seerge** [reply]
Then set "Balanced" in ``Fans + Power`` -> ``Windows Power Mode`` dropdown for all 3 plans, app will never change it. But again - problem is on your windows side. Its not normal  that system behaves like that.

**Reply by steverand** [reply]
Yeah it's balanced and never changes from this. I'll see if I can get anything more concrete but thought it was interesting that someone else has the same symptoms and I thought it could be related. I'll let you know. Thanks for the help =-]

**Reply by seerge** [reply]
Just in case app doesn’t change power modes when your laptop goes to sleep anyway.

It changes it only when you select different mode / preset :)

**Reply by steverand** [reply]
Yep. Everything is balanced everywhere and always has been. =-]

**Reply by joe-700** [reply]
For what its worth; I had an issue with a 2022 G14 not going to sleep properly. Sounds like exactly the same issue your having, it would stay on a solid white light and never properly go to sleep, it even blue screen crashed on wake up a couple of times. I didn't have G-Helper installed on that machine and actually solved it by doing extensive driver updates. Not sure what driver was causing the problem because I used a 3rd party driver updater tool (driver easy) that did a wholesale update of everything.

Sorry I cant be more help. Might be worth rolling back windows installed drivers one at a time to see if you can find the culprit.

**Reply by steverand** [reply]
I nearly went down that path with internet suggestions even talking about wireless mouse USB dongles causing issues. I didn't do drivers or updates but in my case, the Windows Event Logs showed it attempting to go into Windows Modern Standby as the last action... then nothing until a force restart.
powercfg /a showed my laptop doesn't support it, so that's where I looked next. Currently working just fine.

**Reply by seerge** [reply]
Given that different power modes (except balanced) seem to have effect on your sleep, I would look for a trouble in power plans. Did you try to reset power plan to all defaults ? And then probably disable all sleeping wake up timers / etc ?

**Reply by steverand** [reply]
To be clear. I *only* use balanced but I did test with other modes. *all* modes crashed trying to enter a sleep mode not supported by my laptop.

**Reply by seerge** [reply]
Modern stanby is S0 just in case, i think most laptops support S0. 
The one usually not supported is old normal sleep ... S3

**Comment by CONDUCTOR77747** [comment]
I have similar problem. I can't access sleeping mode when G-Helper Ultimate mode is on (Routes laptop screen to dGPU). But if Standard mode (iGPU + dGPU auto mode) is on - everything is ok with sleeping mode. Also, switching between Standard mode and Ultimate mode requires reboot. So annoyning!
I've tried uninstalling Asus Armory Crate. No changes. I'm using G-Helper last release 0.202.0 (Jan 14, 2025). Laptop - ASUS TUF Gaming A15 FA507NV-LP058.

**Reply by steverand** [reply]
i've stayed on 0.148.0.0 and had no further issues. It felt like any time I updated to a newer version it would stop sleeping and i'd go through a hundred things to fix it.
Looking back, it's been a full year since that last post and i've not had a single issue on 0.148.0.0

**Comment by Scarlaid** [comment]
have you done a `sleepstudy` check? 
back then I had issue with the Matrix not doing its sleep animation even on both AC and G-Helper (#3369), Seerge recommended me doing a simple `sleepstudy` check and it would just show what preventing it to sleep, and it was AMD Autio Bus driver thingie, wild

dunno if related, but worth the try to do👍

**Reply by steverand** [reply]
Thanks mate. I did not at the time, just the windows logs.
Running it now shows my audio as always being active, but it sleeps and everything just fine... as long as I don't update Ghelper it seems fine.

**Reply by seerge** [reply]
Unless `powercfg.exe /SleepStudy` explicitly says that it's GHelper.exe preventing windows from entering into sleep - it's unrelated.
